src/include/common/file_system/virtual_file_system.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "file_system.h"
+
+namespace kuzu {
+namespace common {
+
+class VirtualFileSystem final : public FileSystem {
```

Comment:
Just to document the idea here. 
The concept of `VirtualFileSystem` is not very clear to me, as that it is a subclass of `FileSystem`, while actually serving as a proxy to different implementations of `FileSystem`. Alternatively, we can let the `VirtualFileSystem` not extend `FileSystem`, but purely as a container or proxy of all registered `FileSystem`s, and it will be simplified to only provide a `findFileSystem` function. Additionally, we will need a `FileSystemUtils` class to simplify the callers' life by encapsulating the call chain, like `findFileSystem(path)->readFromFile(...)`
I don't see the harm of the alternative way, and it looks like cleaner to me. though the current one is more flexible in terms of usages.